### PR TITLE
4 database connection

### PR DESCRIPTION
**Pull request by Asman97**
merge conflicts resolved; 
parameters, validation and relationships were added to entities, accordingly repositories were changed;
tried to save data from submit;

LAST TASK FROM DATABASE CONNECTION: UNIT TESTING HAS NOT BEEN YET DONE. For task completion, I'll do JUnit testing course in CodeAcademy.

### CODE REVIEW

**Comment by dEisenbeiss** [comment]
Application does not start. Stops with errors! 
This must be fixed

**Comment by Asman97** [comment]
application shall work now

File: src/main/java/com/example/gradecalculator/entities/GradeType.java [hunk 1]
```diff
@@ -3,38 +3,59 @@
...
     private Long id;
-
-    @Column(name = "name")
+    @Column( unique = true)
     private String name;
 
-    public GradeType() {
-    }
+    private String description;
+    private int weightage;
```
Comment (dEisenbeiss) [inline src/main/java/com/example/gradecalculator/entities/GradeType.java:16]: Is it best to use an INT in this case? Weightage of a grade can only be a multiple of itself like x * 1, x * 2, but edge-cases like the grade only counts as half a "normal" grade would not be feasbil. 
Consider using a data type that offers more variety

File: src/main/java/com/example/gradecalculator/entities/Subject.java [hunk 2]
```diff
@@ -3,38 +3,52 @@
...
 
-    @Column(name = "name")
+    @Column(unique = true)
     private String name;
 
-    public Subject() {
-    }
+    private String description;
+
+    //private int creditValue;
```
Comment (dEisenbeiss) [inline src/main/java/com/example/gradecalculator/entities/Subject.java:18]: please remove commented out code as it is not used

Comment (Asman97) [reply src/main/java/com/example/gradecalculator/entities/Subject.java:18]: I wanted first ask if we need this parameter but I found out just know that we need it for calculation of GPA. Thus I will remove commenting.

File: src/main/java/com/example/gradecalculator/entities/GradeType.java (hunk 1 above)
Comment (Asman97) [reply src/main/java/com/example/gradecalculator/entities/GradeType.java:16]: You are totally right, I would change it to percentage.

**Comment by Asman97** [comment]
Requested changes have been implemented

File: pom.xml
[the remark this answers is not in the document]
Comment (dEisenbeiss) [reply pom.xml:1]: @Asman97  there were no changes for this file; please use CTRL+ALT+l or whatever the MAC equvivalent would be for auto-intend-lines

[the remark this answers is not in the document]
Comment (Asman97) [reply pom.xml:1]: I'm sorry, I overlooked. It's done

**Review by dEisenbeiss — APPROVED** [review]

**Comment by dEisenbeiss** [comment]
As a first this is OK, surley needs some refactoring later

**Pull request merged**